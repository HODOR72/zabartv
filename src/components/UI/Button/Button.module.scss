.btn {
	display: inline-flex;
	justify-content: center;
	align-items: center;
	text-transform: uppercase;
	font-weight: 700;
	color: #fff;
	letter-spacing: 0.05em;
	font-size: 20px;
	line-height: 26px;
	background-color: var(--color-primary);
	padding: 21px 30px 19px;
	border-radius: 50px;
	transition: background-color 0.3s, color 0.3s;

	&[disabled] {
		pointer-events: none;
	}

	&:hover {
		background-color: var(--color-primary-active);
	}
}

.gradient {
	background: linear-gradient(314.12deg, #efda1d -47.81%, #fca11a 67.79%);

	&:hover {
		background: linear-gradient(319.22deg, #efda1d 13.05%, #fca11a 82.02%);
	}
}

.small {
	font-size: 16px;
	line-height: 28.8px;
	padding: 14px 20px 12px;
	min-width: 276px;
}

.icon {
	display: block;
	width: 25px;
	height: 27px;
	margin-right: 7px;

	svg {
		display: block;
		width: 100%;
		height: 100%;
		fill: currentColor;
		stroke: currentColor;
	}
}

.white {
	background-color: #fff;
	color: var(--color-primary);

	&:hover {
		color: #fff;
		background-color: var(--color-primary);
	}
}

[data-theme='light'] {
	.dark {
		background-color: #ecf1fa;
		color: #7e8bb4;

		&:hover {
			background-color: #dde7fb;
			color: #7e8bb4;
		}
	}
}

.dark {
	background-color: #141f34;
	color: var(--color-primary);

	svg {
		fill: currentColor;
	}

	&:hover {
		background-color: #212b3b;
		color: var(--color-primary-active);
	}
}

[data-theme='light'] {
	.stroke {
		border: 1px solid #dadee7;

		&:hover {
			background-color: #dadee7;
		}
	}
}

.stroke {
	border: 1px solid rgba(#fff, 0.2);
	background-color: transparent;
	color: var(--color-white);

	@media (max-width: 576px) {
		font-size: 16px;
		line-height: 28.8px;
		padding: 16px 30px 13px;
	}

	span svg {
		stroke: none;
	}

	&:hover {
		background-color: rgba(#fff, 0.2);
		color: #fff;
	}
}
