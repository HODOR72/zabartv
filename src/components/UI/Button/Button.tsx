import {FC, forwardRef, PropsWithChildren, ReactNode} from 'react';
import {ButtonBase} from '@/components/ButtonBase/ButtonBase';
import classNames from 'classnames';
import styles from './Button.module.scss'

export interface ButtonProps {
    gradient?: boolean;
    icon?: ReactNode;
    size?: 'large' | 'medium' | 'small';
    as?: 'link' | 'button';
    className?: string
}

export const Button: FC<PropsWithChildren<ButtonProps>> = forwardRef(({as = 'button', children, gradient, icon, size = 'medium', className, ...props}, ref) => {

    const proxy = {
        'link': 'a',
        'button': ButtonBase
     }

    const Element = proxy[as]

    return (
        <Element
            className={classNames(
                styles.btn,
                gradient && styles.gradient,
                size === 'large' && styles.large,
                size === 'medium' && styles.medium,
                size === 'small' && styles.small,
                className
            )}
            {...props}
        >
            {icon && <span className={styles.icon}>{icon}</span>}
            {children}
        </Element>
    )
})

Button.displayName = 'Button.tsx'