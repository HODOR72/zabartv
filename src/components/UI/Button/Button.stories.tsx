import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Button } from './Button';

export default {
	title: 'Button',
	component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Default = Template.bind({});
Default.args = {
	children: 'Button text',
};

export const Stroke = Template.bind({});
Stroke.args = {
	children: 'Button text',
	variant: 'stroke',
};

export const Dark = Template.bind({});
Dark.args = {
	children: 'Button text',
	variant: 'dark',
};

export const Gradient = Template.bind({});
Gradient.args = {
	children: 'Button text',
	variant: 'gradient',
};

export const White = Template.bind({});
White.args = {
	children: 'Button text',
	variant: 'white',
};
