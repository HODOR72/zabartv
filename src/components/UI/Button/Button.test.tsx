import { Button } from './Button';
import { render, screen } from '@testing-library/react';

test('Button.test.tsx', () => {
	render(<Button>Button text</Button>);
	const element = screen.getByTestId('button');
	expect(element).toBeInTheDocument();
	expect(element.tagName).toEqual('BUTTON');
	expect(element).toMatchSnapshot();
});
