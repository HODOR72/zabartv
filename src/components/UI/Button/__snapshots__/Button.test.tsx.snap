// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`кнопка клик на кнопку 1`] = `
<DocumentFragment>
  <button
    class="btn btn medium"
    data-testid="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`кнопка разные тесты 1`] = `
<button
  class="btn btn medium"
  data-testid="button"
>
  Button text
</button>
`;
